lib/ansible/modules/cloud/ovirt/ovirt_templates.py

```diff
@@ -199,25 +206,38 @@ def post_import_action(self, entity):
         self._service = self._connection.system_service().templates_service()
 
 
-def wait_for_import(module, templates_service):
+def wait_for_import(module, templates_module, templates_service):
     if module.params['wait']:
         start = time.time()
         timeout = module.params['timeout']
         poll_interval = module.params['poll_interval']
         while time.time() < start + timeout:
-            template = search_by_name(templates_service, module.params['name'])
+            template = _fetch_template(module, templates_module, templates_service)
             if template:
                 return template
             time.sleep(poll_interval)
 
 
+def _fetch_template(module, templates_module, templates_service):
+    if (module.params['id'] is not None):
+        # The template id can not be searched with templates_module.search_entity
+        # since the 'id' attribute is not searchable in the engine, therefore searching
+        # the id through templates_service.template_service.
```

Comment:
Yes, I thought about this also, but haven't managed to post a patch of it yet.
I will give it a try